src/server/services/procedures/new-york-public-library/new-york-public-library.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const ApiConsumer = require('../utils/api-consumer');
+const {NewYorkPublicLibraryKey} = require('../utils/api-key');
+const NYPL = new ApiConsumer('NewYorkPublicLibrary', 'http://api.repo.nypl.org/api/v1', {cache: {ttl: 5*60}});
+ApiConsumer.setRequiredApiKey(NYPL, NewYorkPublicLibraryKey);
+
+function listify(item) {
+    return item === undefined ? [] : item instanceof Array ? item : [item];
+}
+
+/**
+ * Searches the NYPL db an returns up to count ids denoting matching items.
+ * 
+ * @param {String} term Term to search for
+ * @param {BoundedNumber<1>} per_page Maximum number of items to return for any given page
+ * @param {BoundedNumber<1>} page Page number of results to get
+ * @returns {Array} Up to per_page matching entries, each being [uuid, title, digitized date, item id, image id]
+ */
+NYPL.search = function(term, per_page, page) {
+    return this._requestData({
+        path:'items/search',
+        queryString:`q=${term}&per_page=${per_page}&page=${page}&publicDomainOnly=true`,
+        headers:{Authorization:`Token token=${this.apiKey.value}`},
+    }).then(res => {
+        const results = res.nyplAPI.response.result;
+        if (results === undefined) return [];
+
+        const ret = [];
+        for (const item of listify(results)) {
+            const itemURL = item.apiItemURL;
+            const itemID = itemURL.substr(itemURL.lastIndexOf('/') + 1);
+            ret.push([item.uuid, item.title, item.dateDigitized, itemID, item.imageID]);
```

Comment:
Actually, if you use `imageID` as the argument for `getImage`, it might be nice to use camelcase here, too:
```javascript
ret.push({
  ID: itemID,
  title: item.title,
  dateDigitized: item.dateDigitized,
  imageID: item.imageID,
});
```

Reply:
`itemID` can be used to generate the image URL directly without going through another API call, which is what the code used to do. Unfortunately, it also requires an image format type, which is not guaranteed to exist, so in the end I just rewrote it to do the API call anyway. So it's not used anymore - I'll remove it for clarity.

Reply:
I think `uuid` and `itemID` are artifacts of some older version. The `itemID` is used for the `/items/` API call, while the `uuid` is used for everything else, like `/items/mods/`.